feat: Log query timing with suspense waterfall detection (#699)
Shopify/hydrogen: packages/hydrogen/src/utilities/log/docs/3-options.md

> Logger has options to turn on/off certain log type informations:

Reviewer: ```suggestion
Logger has the following Boolean options:
```

> | Log options              | Description                                                                          |

Reviewer: ```suggestion
| Option              | Description                                                                          |
```

| ------------------------ | ------------------------------------------------------------------------------------ |

> | `showCacheApiStatus`     | Logs the cache status of each stored entries (PUT/HIT/MISS/STALE)                    |
> | `showCacheControlHeader` | Logs the cache control headers of the main document and its sub queries              |
> | `showQueryTiming`        | Logs the timeline of when queries are being loaded, requested, fetched, and rendered |

Reviewer: ```suggestion
| `showCacheApiStatus`     | Logs the cache status of each stored entry: `PUT`, `HIT`, MISS` or `STALE`.                    |
| `showCacheControlHeader` | Logs the cache control headers of the main document and its sub queries.              |
| `showQueryTiming`        | Logs the timeline of when queries are being loaded, requested, fetched, and rendered. |
```
